src/tool.rs

```diff
@@ -207,25 +214,30 @@ impl Tool {
                     &compiler_detect_output,
                 )?;
                 let stdout = String::from_utf8_lossy(&stdout);
-                guess_family_from_stdout(&stdout, path, cargo_output)
+                guess_family_from_stdout(&stdout, path, args, cargo_output)
             } else {
-                guess_family_from_stdout(&stdout, path, cargo_output)
+                guess_family_from_stdout(&stdout, path, args, cargo_output)
             }
         }
-        let detect_family = |path: &Path| -> Result<ToolFamily, Error> {
-            if let Some(family) = cached_compiler_family.read().unwrap().get(path) {
+        let detect_family = |path: &Path, args: &[String]| -> Result<ToolFamily, Error> {
+            let cache_key = [path]
+                .iter()
+                .map(|p| p.as_os_str().to_owned())
+                .chain(args.iter().map(|a| OsStr::new(a).to_owned()))
+                .collect();
```

Comment:
```suggestion
            let cache_key = [path.as_os_str()]
                .iter()
                .chain(args.iter().map(OsStr::new)
                .map(|s| s.to_owned().into_boxed_slice())
                .collect();
```

Reply:
🤔 This suggestion doesn't compile, but I was able to make this compile:

```suggestion
            let cache_key = [path.as_os_str()]
                .iter()
                .cloned()
                .chain(args.iter().map(osstr::new))
                .map(into::into)
                .collect();
```

Pushed; interdiff link: https://github.com/rust-lang/cc-rs/compare/17c805a543f09e60f8c25296f322c9dd12c319bd..5d2908124311708917ceee5a59a4dfe01f35d5ff